src/main/scala/org/ergoplatform/nodeView/history/ErgoHistoryReader.scala

```diff
@@ -266,6 +266,26 @@ trait ErgoHistoryReader
     }
   }
 
+  /**
+    * Finding other peer's continuation header from a header that is common to our node's history
+    * @param syncInfo  other's node sync info
+    * @return maybe continuation header
+    */
+  def continuationHeaderV2(syncInfo: ErgoSyncInfoV2): Option[Header] = {
+    if (syncInfo.lastHeaders.isEmpty) {
+      Option.empty
+    } else {
+      // let's find most recent header that is common to our history
+      commonPoint(syncInfo.lastHeaders) match {
+        case Some(commonHeader) =>
+          // now let's find continuation header whose parent is the common header
+          syncInfo.lastHeaders.find(_.parentId == commonHeader.id)
```

Comment:
In the real network (i.e. unlike tests) headers in syncv2 are not consecutive (they are located at offsets 0,16,128,512 from the chain tip), so this code will never succeed 

I think should be like

```
  if (syncInfo.lastHeaders.isEmpty) {
      Option.empty
    } else {
      val lastHeader = syncInfo.lastHeaders.head
      if (bestHeaderIdOpt.contains(lastHeader.parentId)) {
        Some(lastHeader)
      } else {
        Option.empty
      }
    }
```

Reply:
@kushti I was under the impression that the `syncV2` headers would rarely begin with our node's `bestHeaderIdOpt` so I was trying to find the `commonPoint` and apply header whose parent is the common point.